.pre-commit-config.yaml

```
@@ -0,0 +1,10 @@
+repos:
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  # Ruff version.
+  rev: v0.4.10
```

Could we possibly use master here?
```suggestion
  rev: master
```

---

I think we should match requirements.txt and keep it pinned